compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -420,11 +428,9 @@ impl TypeAlias {
     }
 
     /// True if the given index is the same index as a generic type of this alias
-    /// which is expected to be a numeric generic.
-    /// This is needed because we infer type kinds in Noir and don't have extensive kind checking.
+    /// which is a numeric generic.
     pub fn generic_is_numeric(&self, index_of_generic: usize) -> bool {
-        let target_id = self.generics[index_of_generic].type_var.id();
-        self.typ.contains_numeric_typevar(target_id)
+        self.generics[index_of_generic].is_numeric()
```

Comment:
Assuming this other `generic_is_numeric` can be removed if Maxim says the first one can. Just noting it here.